src/vs/workbench/contrib/externalTerminal/browser/externalTerminal.contribution.ts

```diff
@@ -20,119 +20,147 @@ import { IRemoteAgentService } from 'vs/workbench/services/remote/common/remoteA
+			const terminalGroupService = accessor.get(ITerminalGroupService);
+			let externalTerminalService: IExternalTerminalService | undefined = undefined;
+			try {
+				externalTerminalService = accessor.get(IExternalTerminalService);
+			} catch {
+			}
+
+			const resources = getMultiSelectedResources(resource, accessor.get(IListService), editorService, accessor.get(IExplorerService));
+			return fileService.resolveAll(resources.map(r => ({ resource: r }))).then(async stats => {
+				// Always use integrated terminal when using a remote
+				const config = configurationService.getValue<IExternalTerminalConfiguration>();
+				const useIntegratedTerminal = remoteAgentService.getConnection() || explorerKind === 'integrated';
+				const targets = distinct(stats.filter(data => data.success));
+				if (useIntegratedTerminal) {
+					// TODO: Use uri for cwd in createterminal
+					const opened: { [path: string]: boolean } = {};
+					const cwds = targets.map(({ stat }) => {
+						const resource = stat!.resource;
+						if (stat!.isDirectory) {
+							return resource;
+						}
+						return URI.from({
+							scheme: resource.scheme,
+							authority: resource.authority,
+							fragment: resource.fragment,
+							query: resource.query,
+							path: dirname(resource.path)
+						});
 					});
-				});
-				for (const cwd of cwds) {
-					if (opened[cwd.path]) {
-						return;
-					}
-					opened[cwd.path] = true;
-					const instance = await integratedTerminalService.createTerminal({ config: { cwd } });
-					if (instance && instance.target !== TerminalLocation.Editor && (resources.length === 1 || !resource || cwd.path === resource.path || cwd.path === dirname(resource.path))) {
-						integratedTerminalService.setActiveInstance(instance);
-						terminalGroupService.showPanel(true);
+					for (const cwd of cwds) {
+						if (opened[cwd.path]) {
+							return;
+						}
+						opened[cwd.path] = true;
+						const instance = await integratedTerminalService.createTerminal({ config: { cwd } });
+						if (instance && instance.target !== TerminalLocation.Editor && (resources.length === 1 || !resource || cwd.path === resource.path || cwd.path === dirname(resource.path))) {
+							integratedTerminalService.setActiveInstance(instance);
+							terminalGroupService.showPanel(true);
+						}
 					}
+				} else if (externalTerminalService) {
+					distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
+						externalTerminalService!.openTerminal(config.terminal.external, cwd);
+					});
 				}
-			} else if (externalTerminalService) {
-				distinct(targets.map(({ stat }) => stat!.isDirectory ? stat!.resource.fsPath : dirname(stat!.resource.fsPath))).forEach(cwd => {
-					externalTerminalService!.openTerminal(config.terminal.external, cwd);
-				});
-			}
-		});
-	}
-});
+			});
+		}
+	});
+}
+
+registerOpenTerminalCommand(OPEN_IN_TERMINAL_COMMAND_ID, 'integrated');
```

Comment:
@Tyriar The command is kept for compatibility with historical bindings, but one thing is not perfect, if the previous user binds the command to 'external', it will be invalid after the update, and the current registration method cannot bind multiple choices to one id, some headache